lib/firebase_functions_interop.dart

```diff
@@ -378,3 +381,51 @@ class FirestoreEvent extends Event<DeltaDocumentSnapshot> {
           timestamp: timestamp,
         );
 }
+
+class PubsubFunctions {
+  const PubsubFunctions._();
+
+  TopicBuilder topic(String path) => new TopicBuilder._(_js.pubsub.topic(path));
+}
+
+class TopicBuilder {
+  @protected
+  final js.TopicBuilder nativeInstance;
+
+  TopicBuilder._(this.nativeInstance);
+
+  /// Event handler that fires every time an event is public in Pubsub.
+  js.CloudFunction onPublish(FutureOr<void> handler(Message message)) {
```

Comment:
According to docs the handler should expect `Event<Message>` similar to other services, right?

I recommend creating `PubsubEvent extends Event<Message>` subclass here (similar to `FirestoreEvent` and `DatabaseEvent`) and update this function accordingly.

JS bindings might need the same changes I suspect.

Reply:
You're right. I started by coding it but couldn't find the doc for it. Found it now though! Gonna fix it.

Reply:
Cool, just in case, I've been looking at this section: https://firebase.google.com/docs/reference/functions/functions.pubsub